testing/deployer/topology/topology.go

```diff
@@ -837,14 +866,38 @@ func (s *Service) Validate() error {
 			return fmt.Errorf("cannot specify both singleport and multiport on service in v2")
 		}
 		if s.Port > 0 {
-			s.Ports = map[string]int{"legacy": s.Port}
+			s.Ports = map[string]*Port{
+				"legacy": {
+					Number:   s.Port,
+					Protocol: "tcp",
+				},
+			}
 			s.Port = 0
 		}
 
+		if !s.DisableServiceMesh && s.EnvoyPublicListenerPort > 0 {
```

Comment:
I just moved this automatic injection to where it more naturally worked given the other changes.